test/ng/compileSpec.js

```diff
@@ -4033,6 +4033,68 @@ describe('$compile', function() {
     });
 
 
+    it('should bind to multiple directives controllers via object notation (new scope)', function() {
+      var controllerCalled = false;
+      var secondControllerCalled = false;
+      module(function($compileProvider, $controllerProvider) {
+        $controllerProvider.register('myCtrl', function() {
+          expect(this.data).toEqualData({
+            'foo': 'bar',
+            'baz': 'biz'
+          });
+          expect(this.str).toBe('Hello, world!');
+          expect(this.fn()).toBe('called!');
+          controllerCalled = true;
+        });
+        $controllerProvider.register('secondCtrl', function() {
+          expect(this.data).toEqualData({
+            'foo2': 'bar2',
+            'baz2': 'biz2'
+          });
+          expect(this.str).toBe('Hello, second world!');
+          expect(this.fn()).toBe('second called!');
+          secondControllerCalled = true;
+        });
+        $compileProvider.directive('fooDir', valueFn({
+          bindToController: {
+            'data': '=dirData',
+            'str': '@dirStr',
+            'fn': '&dirFn'
+          },
+          scope: true,
+          controller: 'myCtrl as myCtrl'
+        }));
+        $compileProvider.directive('barDir', valueFn({
+          bindToController: {
+            'data': '=barData',
+            'str': '@barStr',
+            'fn': '&barFn'
+          },
+          scope: true,
```

Comment:
technically, you're not supposed to have multiple "new scope" directives on the same element, so this doesn't really work in practice (if it does work, it's a bug!).

I'm a bit uncomfortable with this because, given the above, it could become unclear what's happening if you have directives which implicitly depend on there being a new scope directive on the same element to work correctly.